### PR TITLE
GUVNOR-2565: NullPointerException when deleting repository using REST API

### PR DESCRIPTION
**Pull request by dgutierr**
Fix + test

@mswiderski Can you verify please?

### CODE REVIEW

**Comment by manstis** [comment]
+1 to merge.

File: guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/organizationalunit/OrganizationalUnitServiceImpl.java [hunk 1]
```diff
@@ -114,6 +114,11 @@ public OrganizationalUnit getOrganizationalUnit( final String name ) {
 
     @Override
     public Collection<OrganizationalUnit> getAllOrganizationalUnits() {
+        return new ArrayList<>( registeredOrganizationalUnits.values() );
+    }
+
+    @Override
+    public Collection<OrganizationalUnit> getOrganizationalUnits() {
```
Comment (psiroky) [inline guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/organizationalunit/OrganizationalUnitServiceImpl.java]: Could we use more descriptive name? When looking at the code, it is not obvious to me what is the difference between `getAllOrganizationalUnits()` and `getOrganizationalUnits()`. Perhaps something like `getAccessibleOrganizationUnits`?

**Comment by psiroky** [comment]
Verified that this fixes the currently failing REST tests in `kie-wb-distributions/kie-wb-tests/kie-wb-tests-rest`.

File: guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/organizationalunit/OrganizationalUnitServiceImpl.java (hunk 1 above)
Comment (dgutierr) [inline guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/organizationalunit/OrganizationalUnitServiceImpl.java]: I tried not to break API compatibility, that's why I kept the same method name. Anyway, provided javadoc is quite self-explanatory.

Comment (psiroky) [inline guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/organizationalunit/OrganizationalUnitServiceImpl.java]: Ah! I got confused by the diff, sorry. (now I also see the javadoc in the interface).

**Comment by psiroky** [comment]
+1, merging as it is needed to fix currently failing REST tests.

**Pull request merged**